modules/apps/mobile-device-rules/mobile-device-rules-web/src/main/java/com/liferay/mobile/device/rules/web/internal/display/context/MDRActionDisplayContext.java

```diff
@@ -88,8 +92,18 @@ public String getDisplayStyle() {
 			return _displayStyle;
 		}
 
-		_displayStyle = ParamUtil.getString(
-			_renderRequest, "displayStyle", "list");
+		_displayStyle = ParamUtil.getString(_renderRequest, "displayStyle");
+
+		PortalPreferences portalPreferences =
+			PortletPreferencesFactoryUtil.getPortalPreferences(_renderRequest);
+
+		if (Validator.isNull(_displayStyle)) {
+			_displayStyle = portalPreferences.getValue(
+				MDRPortletKeys.MOBILE_DEVICE_RULES, "display-style", "list");
+		}
+
+		portalPreferences.setValue(
```

Comment:
Shouldn't this be inside a "else"? If you were just inside the "if (Validator.isNull(_displayStyle)) {" you just got the "_displayStyle" from portalPreferences, why would you want to put the same value back in?

Reply:
:thinking: That's changed in the final commits. Perhaps GitHub is playing with your display and missed some commits?

Reply:
Oh, ok, I was just given this single commit to review, I did not check the whole pull.

Reply:
Thanks @shuyangzhou, it was a miscommunication :worried:

Reply:
:+1: No worries. GitHub has these things...